### ISSUE TITLE
'property' is missing props validation, false positive when passing props to function declared on stateless component

### ISSUE THREAD

**Issue by nc-babelv** [yannickcr/eslint-plugin-react#1003]
eslint version: `2.1.0`
eslint-plugin-react version: `5.2.2`

I am running into an issue linting stateless functional components. I get a false positive for the `prop-types` rule when I pass the `props` object to a function declared on the functional component that returns a node.

For example:
```js
import React from 'react';

import OtherComponent from 'other-component';

const TestComponent = (props) => {
  return (
    <OtherComponent
      detail={ TestComponent.config.getDetail(props) }
    />
  );
};

TestComponent.propTypes = {
  propOne: React.PropTypes.number.isRequired,
  propTwo: React.PropTypes.number.isRequired
};

TestComponent.config = {
  getDetail({ propOne, propTwo }) {
    return <div>{ propOne + propTwo }</div>;
  }
};

export default TestComponent;
```

What's bizarre is, if `TestComponent.config.getDetail` doesn't return a `div`, and instead just returns some operation on the props, I do not get the `prop-types` lint, ie:
```js
getDetail({ propOne, propTwo }) {
  return propOne + propTwo;
}
```

Not sure what is happening here, and I can get around it if I do the above and wrap the result of the function in a `div` within the component, like so:
```js
const TestComponent = (props) => {
  return (
    <OtherComponent
      detail={ <div>{ TestComponent.config.getDetail(props) }</div> }
    />
  );
};
```

**Comment by ljharb** [comment]
A function that returns jsx is treated as a component. In this case, `getDetail` should be an SFC, not just a jsx-returning function.

I'm also not sure why you're sticking those functions directly on the TestComponent as opposed to just referencing them lexically in the closure?

**Comment by nc-babelv** [comment]
Ah ok, that makes sense, thanks.  We put functions like that on the TestComponent for configurability, so it seems like that function should just be returning just the content, and not the element thats wrapping it.  Thanks again!
